lib/Controller/SocialApiController.php

```diff
@@ -49,6 +49,40 @@ class SocialApiController extends ApiController {
 	/** @var IConfig */
 	private  $config;
 
+	/**
+	 * This constant stores the supported social networks
+	 * It is an ordered list, so that first listed items will be checked first
+	 * Each item stores the avatar-url-formula as recipe, a cleanup parameter to
+	 * extract the profile-id from the users entry, and possible filters to check
+	 * validity
+	 * 
+	 * @const {array} SOCIAL_CONNECTORS dictionary of supported social networks
+	 */
+	const SOCIAL_CONNECTORS = [
+		'facebook' 	=> [
+			'recipe' 	=> 'https://graph.facebook.com/{socialId}/picture?width=720',
+			'cleanups' 	=> ['basename'],
+			'checks'	=> ['number'],
```

Comment:
Number?
`https://graph.facebook.com/nextclouders/picture?width=720`

Reply:
I think this works only for public pages, normally you will find a profile id of a person here. And if you want to get the graphic for those, you need to authenticate with facebook first.

Reply:
It can be a company vCard :)

Reply:
> I am still wondering how to tell users its working for ids but not for names.

But it is working for names ^^
My link above works fine :)
![](https://graph.facebook.com/nextclouders/picture?width=720)

Reply:
just making sure: for most contacts it will only work with the numbered id
https://graph.facebook.com/karlitschek/picture?width=720 vs.
https://graph.facebook.com/1004752406/picture?width=720

only for public pages it works additionally with the name:
https://graph.facebook.com/nextclouders/picture?width=720
but also with the id:
https://graph.facebook.com/1032807203462807/picture?width=720

This is why I initially checked if the id is a number. I suppose that most of the contacts are people and not public.

Reply:
Good to know! :)
We could improve later I think